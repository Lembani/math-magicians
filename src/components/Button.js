import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { val, handleChange, styleClass } = props;
  return (
    <button onClick={handleChange} className={styleClass} type="button">
      {val}
    </button>
  );
};

Button.propTypes = {
  val: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  styleClass: PropTypes.string.isRequired,
};

export default Button;
