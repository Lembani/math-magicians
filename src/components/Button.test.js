import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from './Button';

test('Renders `Button` to the screen', () => {
  render(<Button />);
  const btnElement = screen.getByRole('button');
  expect(btnElement).toBeInTheDocument();
});

it('Should render `Button` correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
