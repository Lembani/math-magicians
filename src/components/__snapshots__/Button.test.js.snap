// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render \`Button\` correctly 1`] = `
<button
  type="button"
/>
`;
